### PR TITLE
Prepare for browser tests

### PR DESCRIPTION
**Pull request by vweevers**
For early review. Tests are passing locally in Chrome, IE10 and IE11. Yet to do:

- [x] Address TODO comments in `test/inject-encoding-test.js`
- [ ] Add airtap (~~pending Sauce Labs activation~~)
- [ ] Update `abstract-leveldown` (pending https://github.com/Level/abstract-leveldown/pull/221)
- [x] Rebase on master
- [x] Simplify `.travis.yml` (no need to install compiler etc)

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -2,9 +2,29 @@
...
+require('./init-test')
+require('./inject-encoding-test')
+require('./json-test')
+require('./key-value-streams-test')
+require('./maybe-error-test')
+require('./no-encoding-test')
+require('./null-and-undefined-test')
+require('./open-patchsafe-test')
+require('./read-stream-test')
+require('./snapshot-test')
```
Comment (ralphtheninja) [inline test/index.js]: I prefer this because it's more explicit.

File: test/index.js [hunk 2]
```diff
@@ -2,9 +2,29 @@
...
  */
-const fs = require('fs')
-const path = require('path')
 
-fs.readdirSync(__dirname)
-  .filter(f => /-test.js/.test(f))
-  .forEach(f => require(path.join(__dirname, f)))
+// Promise polyfill for IE and others.
+if (process.browser && typeof Promise !== 'function') {
+  global.Promise = require('pinkie')
```
Comment (ralphtheninja) [inline test/index.js]: TIL about pinkie!

File: test/snapshot-test.js [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 var delayed = require('delayed')
 var common = require('./common')
-var SlowStream = require('slow-stream')
+var trickle = require('trickle')
```
Comment (ralphtheninja) [inline test/snapshot-test.js]: @vweevers I take it you ran into some issues with `slow-stream`?

Comment (ralphtheninja) [reply test/snapshot-test.js]: TIL about trickle!

Comment (vweevers) [reply test/snapshot-test.js]: > I take it you ran into some issues with slow-stream?

`slow-stream` uses `process.hrtime` and `setTimeout`, `trickle` only uses `setInterval`.

**Comment by ralphtheninja** [comment]
@vweevers Added a thing. We should simplify `.travis.yml` since we shouldn't need to install compiler etc (which was needed for c++11 stuff).

**Comment by ralphtheninja** [comment]
More precisely, remove these lines:

https://github.com/Level/levelup/blob/78e2d6f03d77749893356fcc43ff62739126b8b4/.travis.yml#L5-L17

File: test/inject-encoding-test.js [hunk 4]
```diff
@@ -64,7 +64,9 @@ buster.testCase('JSON API', {
 
   'simple-object keys in "json" encoding': function (done) {
     this.runTest([
-      { value: '0', key: 0 },
+      // Test keys that would be considered the same with default utf8 encoding.
+      // Because String([1]) === String(1).
+      { value: '0', key: [1] },
```
Comment (vweevers) [inline test/inject-encoding-test.js]: This test was useless because it didn't actually test custom encoding of keys. The keyEncoding was in fact utf8.

**Comment by vweevers** [comment]
If we don't get a response from Sauce Labs soon, I'll revert 505fe10977937d8ef7c10ec098aa076340c3a021, squash, and merge. We can then add airtap + new `abstract-leveldown` in another PR.

**Comment by ralphtheninja** [comment]
@vweevers Think we can get this done during the coming week?

**Comment by vweevers** [comment]
@ralphtheninja yes! I will get back to it monday/tuesday. Not much left to do :)

**Comment by vweevers** [comment]
> If [..] I'll revert 505fe10, squash, and merge. We can then add airtap + new abstract-leveldown in another PR.

Let's do that anyway, this PR is big enough. So @ralphtheninja can you review the last three commits (that I added after you approved)? 757e6d2, 7e978e8 and 55ea183.

I'll then squash (by hand because I want to keep some commits).

**Comment by ralphtheninja** [comment]
LGTM

**Pull request merged**